modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Currency.scala

```diff
@@ -0,0 +1,29 @@
+package org.tessellation.rosetta.domain
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+
+import derevo.circe.magnolia._
+import derevo.derive
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.types.numeric.NonNegLong
+import io.circe.refined._
+import io.estatico.newtype.macros.newtype
+
+object currency {
+  @derive(customizableDecoder, customizableEncoder)
+  case class Currency(
+    symbol: CurrencySymbol,
+    decimals: CurrencyDecimal
+  )
+
+  type CurrencySymbolRefined = String Refined Equal["DAG"]
+
+  @derive(decoder, encoder)
+  @newtype
+  case class CurrencySymbol(value: CurrencySymbolRefined)
+
+  @derive(decoder, encoder)
+  @newtype
+  case class CurrencyDecimal(value: NonNegLong)
```

Comment:
should we bind it to 8 decimals for our network?

Reply:
That decision is beyond my pay grade. I don't know the network well enough. I'll leave that to you guys and @ryle-goehausen to decide.

Reply:
Refined the value to 8.

```
type CurrencyDecimalRefined = Long Refined Equal[8L]

  @derive(decoder, encoder)
  @newtype
  case class CurrencyDecimal(value: CurrencyDecimalRefined)
```